deliver/lib/deliver/app_screenshot_validator.rb

```diff
@@ -0,0 +1,103 @@
+    #
+    # @param screenshot [AppScreenshot]
+    # @param errors [Array<Deliver::AppScreenshotValidator::ValidationError>] Pass an array object to add validation errors when detecting errors
+    # @return [Boolean] true if given screenshot is valid
+    def self.validate(screenshot, errors)
+      # Given screenshot will be diagnosed and errors found are accumulated
+      errors_found = []
+
+      validate_screen_size(screenshot, errors_found)
+      validate_device_type(screenshot, errors_found)
+      validate_file_extension_and_format(screenshot, errors_found)
+
+      # Merge errors found into given errors array
+      errors_found.each { |error| errors.push(error) }
+      errors_found.empty?
+    end
+
+    def self.validate_screen_size(screenshot, errors_found)
+      if screenshot.screen_size.nil?
+        errors_found << ValidationError.new(type: ValidationError::INVALID_SCREEN_SIZE,
+                                            path: screenshot.path,
+                                            debug_info: "Actual size is #{get_formatted_size(screenshot)}")
+      end
+    end
+
+    # Checking if the device type exists in spaceship
+    # Ex: iPhone 6.1 inch isn't supported in App Store Connect but need
+    # to have it in there for frameit support
+    def self.validate_device_type(screenshot, errors_found)
+      if !screenshot.screen_size.nil? && screenshot.device_type.nil?
+        errors_found << ValidationError.new(type: ValidationError::UNACCEPTABLE_DEVICE,
+                                            path: screenshot.path,
+                                            debug_info: "Screen size #{screenshot.screen_size} is not accepted",
+                                            to_skip: true)
+      end
+    end
+
+    def self.validate_file_extension_and_format(screenshot, errors_found)
+      extension = File.extname(screenshot.path).delete('.')
+      valid_file_extensions = ALLOWED_SCREENSHOT_FILE_EXTENSION.values.flatten
+      is_valid_extension = valid_file_extensions.include?(extension)
+
+      unless is_valid_extension
+        errors_found << ValidationError.new(type: ValidationError::INVALID_FILE_EXTENSION,
+                                            path: screenshot.path,
+                                            debug_info: "Only #{valid_file_extensions.join(', ')} are allowed")
+      end
+
+      format = FastImage.type(screenshot.path)
+      is_extension_macthed = ALLOWED_SCREENSHOT_FILE_EXTENSION[format] &&
+                             ALLOWED_SCREENSHOT_FILE_EXTENSION[format].include?(extension)
+
+      # This error only appears when file extension is valid
+      if is_valid_extension && !is_extension_macthed
+        errors_found << ValidationError.new(type: ValidationError::FILE_EXTENSION_MISMATCH,
+                                            path: screenshot.path,
+                                            debug_info: "Actual format is #{format}")
```

Comment:
Likewise, here, as a user, I'd like to see "what it is" and "what it should be". Could we suggest how to fix this issue? For instance if the `image.png` has a jpeg format, they should probably rename the file to `image.jpeg`, right? Should we suggest this?

Reply:
Added a suggestion message (and a test case for it)